src/MethodTypeDetector.php

```diff
@@ -41,6 +41,7 @@ public function isTraitMethod(ClassMethod $classMethod, Scope $scope): bool
         }
 
         $extendedMethodReflection = $classReflection->getMethod($classMethod->name->toString(), $scope);
+        // @phpstan-ignore phpstanApi.instanceofAssumption, phpstanApi.class
```

Comment:
PHPStan complains that:
- Although PHPStan\Reflection\Php\PhpMethodReflection is covered by backward compatibility promise, this instanceof assumption might break because it's not guaranteed to always stay the same
- Asking about instanceof PHPStan\Reflection\ResolvedMethodReflection is not covered by backward compatibility promise. The class might change in a minor PHPStan version.
I don't think there is much we can do at this point apart from ignoring these errors for now